variant/normalize.py

```diff
@@ -141,3 +133,139 @@ def add_extension(self, extensions, name, value):
+                dna_sequence_variant_token.reference_sequence
+            )
+            if self._is_token_type(valid_result_tokens,
+                                   'CodingDNASubstitution') or \
+                    self._is_token_type(valid_result_tokens,
+                                        'GenomicSubstitution'):
+                structural_type = 'SO:0001483'
+                ref_allele_seq = dna_sequence_variant_token.ref_nucleotide
+            elif self._is_token_type(valid_result_tokens,
+                                     'CodingDNASilentMutation') or \
+                    self._is_token_type(valid_result_tokens,
+                                        'GenomicSilentMutation'):
+                structural_type = 'SO:0002073'
+                ref_allele_seq = self.get_delins_ref_allele_seq(allele,
+                                                                label)
+            else:
+                structural_type = None
+                ref_allele_seq = None
+        elif delins_token:
+            molecule_context = self._get_molecule_context(
+                delins_token.reference_sequence
+            )
+            structural_type = 'SO:1000032'
+            ref_allele_seq = self.get_delins_ref_allele_seq(allele,
+                                                            label)
+        else:
+            molecule_context = None
+            structural_type = None
+            ref_allele_seq = None
+        return molecule_context, structural_type, ref_allele_seq
+
+    def get_delins_ref_allele_seq(self, allele, label) -> Optional[str]:
+        """Return ref allele seq for transcript.
+
+        :param dict allele: VRS Allele object
+        :param str label: Transcript label
+        :return: Ref seq allele
+        """
+        label = label.split(':')[0]
+        interval = allele['location']['interval']
+        start = interval['start'] + 1
+        end = interval['end']
+
+        if start and end:
+            refseq_list = list()
+            while start <= end:
+                refseq_list.append(self.seqrepo_access.sequence_at_position(
+                    label, start
+                ))
+                start += 1
+            try:
+                return ''.join(refseq_list)
+            except TypeError:
+                pass
+        return None
+
+    def _is_token_type(self, valid_result_tokens, token_type) -> bool:
+        """Return whether or not token_type is in valid_result_tokens.
+
+        :param list valid_result_tokens: Valid token matches
+        :param str token_type: The token's type
+        :return: Whether or not token_type is in valid_result_tokens
+        """
+        for t in valid_result_tokens:
+            if t.token_type == token_type:
+                return True
+        return False
+
+    def _get_instance_type_token(self, valid_result_tokens, instance_type):
+        """Return the tokens for a given instance type.
+
+        :return: A list of tokens
```

Comment:
Shouldn't this only return a single token (rather than a list?)

Reply:
oops. I meant to put params. Updated it!